lib/WUI/nhttp/send_json.cpp

```diff
@@ -0,0 +1,132 @@
+#include "send_json.h"
+#include "handler.h"
+#include "headers.h"
+
+#include <timing.h>
+#include <segmented_json_macros.h>
+#include <http/chunked.h>
+#include <transfers/monitor.hpp>
+
+using transfers::Monitor;
+using namespace json;
+using namespace http;
+
+namespace nhttp::handler {
+
+json::JsonResult EmptyRenderer::renderState(size_t resume_point, json::JsonOutput &output, Empty &state) const {
+    return generator(resume_point, output);
+}
+
+JsonResult TransferRenderer::renderState(size_t resume_point, json::JsonOutput &output, TransferState &state) const {
+    // Note: We allow stale, because we already checked there was a trasnfer running
+    // in prusa_link_api, so even if it ended, we want to report it.
+    auto transfer_status = Monitor::instance.status(true);
+    // Keep the indentation of the JSON in here!
+    // clang-format off
+    JSON_START;
+        JSON_OBJ_START
+            if (transfer_status.has_value()) {
+                if (transfer_status->id != state.transfer_id) {
+                    // if transfer changes mid report, bail out
+                    transfer_status.reset();
+                }
```

Comment:
Damn. I've noticed this issue only now, should have pointed it out earlier :-(.

The `JSON_START` is a switch statement that jumps to the place where it left previously. That is, if you place this guard _inside_ the `JSON_START`, it'll be skipped over. It needs to go outside / before the JSON_START.

I wonder, considering how much JSON rendering we have here if we should somehow explore the C++ coroutines, here we have quite many ugly cornercases and traps :-|.